pkgs/stdenv/generic/check-meta.nix

```diff
@@ -444,7 +445,14 @@ let
     let
       outputs = attrs.outputs or [ "out" ];
     in
-    {
+    lib.attrsets.optionalAttrs (attrs ? src.meta.homepage) {
+      # should point to an http-browsable source tree, if available.
+      # fetchers like fetchFromGitHub set it automatically.
+      # this could be handled a lot easier if we nulled it instead
+      # of having it be undefined, but that wouldn't match the
+      # other attributes.
+      repository = attrs.src.meta.homepage;
+    } // {
```

Comment:
As I said before, we should not rely on setting this automatically.

Since you and @Aleksanaa are arguing for including this, it should be explicitly set by the package maintainer, never inferred.

```suggestion
```

Reply:
Further it should be a list of repositories, since it is very common a repo having backup mirrors.

Reply:
it is clear that you are misunderstanding the point of this field.

it is not for use by update scripts, that is what `src.gitRepoUrl` is for.

> As I said before, we should not rely on setting this automatically.

you say this, but you do not specify why.

please give an example of when setting this automatically would lead to undesired behavior.  several (most?) of the `meta.*` fields are set automatically.

Reply:
NB packagers would have to explicitly set it to null if there's no browsable repo

Reply:
if there's no browsable repo, `src.meta.homepage` will not be set, and thus `meta.repository` will not be set either.

Reply:
> it is clear that you are misunderstanding the point of this field.

It is clear you gave no reason why this field should point to an HTTP-browsable site to begin with.

> 
> it is not for use by update scripts, that is what `src.gitRepoUrl` is for.

This is irrelevant.
First, nothing hinders this field to be useful for other applications besides search.nixos.org. Second, nothing should hinder this field to be used by other applications besides search.nixos.org.

Third, what is the relevance? When I said this should be made for update scripts?
I merely suggested this could be a multi-valued field, since this is a typical occurrence in nature.

> 
> > As I said before, we should not rely on setting this automatically.
> 
> you say this, but you do not specify why.
> 
> please give an example of when setting this automatically would lead to undesired behavior. several (most?) of the `meta.*` fields are set automatically.

Behold!

![erro-crasso](https://github.com/NixOS/nixpkgs/assets/5954806/0454ee6c-f7af-47ef-8a93-b859708d1bdd)

Reply:
> It is clear you gave no reason why this field should point to an HTTP-browsable site to begin with.

it is supposed to be presented to the user via sites like search.nixos.org

therefore it should be a link that can be opened by a browser

> This is irrelevant.

the existance of a similar but semantically different field is extremely relevant.

there is no reason for this field to represent a machine-readable git repository, as there is already another field that represents that.

> First, nothing hinders this field to be useful for other applications besides search.nixos.org.
> Second, nothing should hinder this field to be used by other applications besides search.nixos.org.

yes, that is correct.  any other application that wants a human-readable, HTTP-browseable source code repository can use this field, because that is what this field represents. 

> Third, what is the relevance? When I said this should be made for update scripts?

yes, that is exactly the relevance

> I merely suggested this could be a multi-valued field, since this is a typical occurrence in nature.

yes, while mirror repositories are common, i don't think modeling that is worth the added complexity.
1. most fetchers already handle mirrors in some way
2. having the user pick from a list of mirrors may cause confusion, needlessly clutter the UI, and make using the value more difficult

if you can show me a practical usecase where multiple values would be helpful, i can update this to allow multiple values.

Reply:
some packages have multiple sources, like `trellis`, and thus multiple repositories.

Reply:
fair enough.